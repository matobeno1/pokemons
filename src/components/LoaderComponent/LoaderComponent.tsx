import { FunctionComponent } from "react";
import Image from "next/image";

import { createClassNames } from "../../bem";

const classNames = createClassNames("loader-component");

export const LoaderComponent: FunctionComponent = () => (
    <div className={classNames()}>
        <Image src="/pokeball.svg" layout="fill"/>
    </div>
);
