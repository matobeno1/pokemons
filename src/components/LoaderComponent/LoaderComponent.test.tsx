import React from "react";
import { act, create, ReactTestRenderer } from "react-test-renderer";
import { LoaderComponent } from "./LoaderComponent";

describe(LoaderComponent, () => {
    it("snapshots", async () => {
        let root: ReturnType<typeof create> | undefined;
        act(() => {
            root = create(<LoaderComponent />);
        });
        expect(root?.toJSON()).toMatchSnapshot();
    });
});
