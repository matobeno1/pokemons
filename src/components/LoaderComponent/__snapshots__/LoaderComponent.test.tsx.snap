// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoaderComponent snapshots 1`] = `
<div
  className="loader-component"
>
  <div
    className="loader-component__image"
  >
    <img
      alt="pokeball-loader-image"
      layout="fill"
      src="/pokeball.svg"
    />
  </div>
</div>
`;
